lib/train/transports/azure.rb

```diff
@@ -59,15 +61,17 @@ def platform
         force_platform!('azure', @platform_details)
       end
 
-      def azure_client(klass = ::Azure::Resources::Profiles::Latest::Mgmt::Client)
+      def azure_client(klass = ::Azure::Resources::Profiles::Latest::Mgmt::Client, vault_name = nil)
```

Comment:
I would prefer valut_name to be a hash here so its flex-able if we want to add other options for clients:
```
azure_client(Azure::KeyVault::Profiles::Latest::Mgmt::Client, vault_name: vault1, valut_id: 1234)

def azure_client(klass = ::Azure::Resources::Profiles::Latest::Mgmt::Client, opts = {})
...
  client = Vault.client(opts[:vault_name], @credentials)
end
```